code/modules/admin/verbs/playsound.dm

```diff
@@ -174,6 +183,96 @@
 
 	SSblackbox.record_feedback("tally", "admin_verb", 1, "Play Internet Sound")
 
+/proc/var_internet_sound(client/src, request)
```

Comment:
also, the indentation hurts at times. i get there's a lot of if/else stuff but any other early returns you can add in (or spinning stuff off to other procs) would be so fantastic